### ISSUE TITLE
Using ProblemDetails

### ISSUE THREAD

**Issue by SeanFeldman** [Azure/azure-functions-dotnet-worker#1586]
I want to be able to use [ProblemDetails](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.problemdetails?view=aspnetcore-7.0) with HTTP-triggered functions. While there's no parity with ASP.NET Core, the [middleware integration](https://learn.microsoft.com/en-us/aspnet/core/web-api/handle-errors?view=aspnetcore-7.0) available in ASP.NET Core would not apply. Would there be something OOTB or should it be a custom functions middleware?

**Comment by fabiocav** [comment]
@SeanFeldman indeed, this is not tracked for the parity milestones, but it is something we'll be supporting once ASP.NET Core extensibility is expanded. 

Will leave this item open to track.

**Comment by PureKrome** [comment]
@fabiocav 

> once ASP.NET Core extensibility is expanded.

is there any plans / rough ETA's on this item, occurring?

**Comment by fabiocav** [comment]
@PureKrome no concrete ETA I'm able to share at this point, but we'll communicate that here and add this issue that planning (label, reference) for tracking.

**Comment by PureKrome** [comment]
@fabiocav thanks mate - really, really appreciate all your (and teams) work 👏🏻 🍰

**Comment by jm-atbas** [comment]
Same here. We already use `ProblemDetails` for Minimal APIs but those aren't compatible with Azure Functions. MVC allows converting `IClientErrorActionResult` into `ProblemDetails` when the controller is annotated with `[ApiController]` which is also not an option here. Right now it looks like the functions need to build the `ProblemDetails` themselves.
